modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/unit/BatchEngineUnitProcessorImpl.java

```diff
@@ -233,6 +248,51 @@ private String _getObjectEntryClassName(
 		return className;
 	}
 
+	private boolean _isProcessed(BatchEngineUnit batchEngineUnit) {
+		Bundle bundle = _getBundle(batchEngineUnit);
+
+		if (bundle == null) {
+			return false;
+		}
+
+		try {
+			BatchEngineUnitConfiguration batchEngineUnitConfiguration =
+				batchEngineUnit.getBatchEngineUnitConfiguration();
+
+			java.io.File batchMarkerFile = bundle.getDataFile(
+				com.liferay.petra.string.StringUtil.merge(
+					Arrays.asList(
+						FileUtil.getShortFileName(
+							batchEngineUnit.getDataFileName()),
+						batchEngineUnitConfiguration.getCompanyId(),
+						"processed"),
+					"."));
+
+			String lastModifiedString = String.valueOf(
+				bundle.getLastModified());
+
+			if ((batchMarkerFile != null) && batchMarkerFile.exists() &&
+				Objects.equals(
+					FileUtil.read(batchMarkerFile), lastModifiedString)) {
+
+				return true;
+			}
+
+			if (!batchMarkerFile.exists()) {
```

Comment:
What if `batchMarkerFile` is null? You checked in the previous if clause, and if it is null, then here it would be null. One of the two if clauses is not right

Reply:
Fixed in https://github.com/liferay-headless/liferay-portal/pull/1546/commits/2b7ddb8e78ab664e56e708b35b6079cae28bac26.